File: Imperative/Java/half_of_a_5x5_square/half_of_a_5x5_square.java
// half_of_a_5x5_square.java

public class half_of_a_5x5_square {

	public static void main( String args[]) {

		for ( int n = 5; n > 0; n--) {

			for ( int c = 0; c < n; c++ ) {

				System.out.printf("%c", '#');

			}
			
			System.out.printf("%c", '\n');

		}	

	}

}

File: Imperative/C++/half_of_a_5x5_square.cpp
// half_of_a_5x5_square.cpp

/* Prints out a triangle that
 *
 * looks exactly like this:
 *
 * #####
 * ####
 * ###
 * ##
 * #
 *
 * Conditions:
 *
 * Only allowed to use two
 * output statements:
 *
 * cout << "#";
 *
 * AND
 *
 * cout << "\n";
 *
 */

#include <iostream>
using namespace std;

int main() {

	for ( int n = 5; n > 0; n--) {

		for ( int c = 0; c < n; c++ ) {

			cout << "#";

		}
			
		cout << "\n";
	}

}
